src/ui/display_nbgl.c

```diff
@@ -322,347 +127,460 @@ void ui_display_output_address_amount_flow(int index) {
 
-    transactionContext.tagValuePair[1].item = "Public key";
-    transactionContext.tagValuePair[1].value = g_ui_state.path_and_pubkey.pubkey;
-    transactionContext.tagValueList.nbPairs = 2;
+    contentList[0].type = CENTERED_INFO;
+    contentList[0].content.centeredInfo.icon = &C_Bitcoin_64px;
+    contentList[0].content.centeredInfo.text1 = "Confirm public key";
+    contentList[0].content.centeredInfo.text2 = NULL;
+    contentList[0].content.centeredInfo.text3 = NULL;
+    contentList[0].content.centeredInfo.style = LARGE_CASE_BOLD_INFO;
+    contentList[0].content.centeredInfo.offsetY = 0;
+    contentList[0].contentActionCallback = NULL;
 
-    transactionContext.confirm = "Approve public key";
-    transactionContext.confirmed_status = "PUBLIC KEY\nAPPROVED";
-    transactionContext.rejected_status = "Public key rejected";
+    contentList[1].type = TAG_VALUE_LIST;
+    memcpy(&contentList[1].content.tagValueList, &pairList, sizeof(nbgl_layoutTagValueList_t));
+    contentList[1].contentActionCallback = NULL;
 
-    nbgl_useCaseReviewStart(&C_Bitcoin_64px,
-                            "Confirm public key",
-                            "",
-                            "Cancel",
-                            continue_light_notify_callback,
-                            status_cancel);
+    contentList[2].type = INFO_BUTTON;
+    contentList[2].content.infoButton.text = "Approve public key";
+    contentList[2].content.infoButton.icon = &C_Bitcoin_64px;
+    contentList[2].content.infoButton.buttonText = "Approve";
+    contentList[2].content.infoButton.buttonToken = REVIEW_CONFIRM;
+    contentList[2].content.infoButton.tuneId = TUNE_TAP_CASUAL;
+    contentList[2].contentActionCallback = generic_content_callback;
+
+    genericContent.callbackCallNeeded = false;
+    genericContent.contentsList = contentList;
+    genericContent.nbContents = 3;
+
+    nbgl_useCaseGenericReview(&genericContent, "Cancel", status_cancel);
 }
 
 void ui_display_receive_in_wallet_flow(void) {
-    transactionContext.tagValuePair[0].item = "Wallet name";
-    transactionContext.tagValuePair[0].value = g_ui_state.wallet.wallet_name;
+    confirmed_status = "ADDRESS\nCONFIRMED";
+    rejected_status = "Address rejected";
+
+    // Setup list
+    pairs[0].item = "Wallet name";
+    pairs[0].value = g_ui_state.wallet.wallet_name;
+
+    pairs[1].item = "Wallet Address";
+    pairs[1].value = g_ui_state.wallet.address;
 
-    transactionContext.tagValuePair[1].item = "Wallet Address";
-    transactionContext.tagValuePair[1].value = g_ui_state.wallet.address;
+    // Setup list
+    pairList.nbMaxLinesForValue = 0;
+    pairList.nbPairs = 2;
+    pairList.pairs = pairs;
 
-    transactionContext.tagValueList.nbPairs = 2;
+    contentList[0].type = CENTERED_INFO;
+    contentList[0].content.centeredInfo.icon = &C_Bitcoin_64px;
+    contentList[0].content.centeredInfo.text1 = "Receive\nin known wallet";
+    contentList[0].content.centeredInfo.text2 = NULL;
```

Comment:
All functions calling nbgl_useCaseGenericReview could be factorized in one with only the pairList as parameter and maybe two strings.